sn_cli/src/subcommands/wallet/wo_wallet.rs

```diff
@@ -0,0 +1,443 @@
+    let main_pk = MainPubkey::from_hex(from)?;
+    let mut wallet = watch_only_wallet_from_pk(main_pk, root_dir)?;
+    let amount = match NanoTokens::from_str(amount) {
+        Ok(amount) => amount,
+        Err(err) => {
+            println!("The amount cannot be parsed. Nothing sent.");
+            return Err(err.into());
+        }
+    };
+    let to = match MainPubkey::from_hex(to) {
+        Ok(to) => to,
+        Err(err) => {
+            println!("Error while parsing the recipient's 'to' key: {err:?}");
+            return Err(err.into());
+        }
+    };
+
+    let unsigned_transfer = wallet.build_unsigned_transaction(vec![(amount, to)], None)?;
+
+    println!(
+        "The unsigned transaction has been successfully created:\n\n{}\n",
+        hex::encode(rmp_serde::to_vec(&unsigned_transfer)?)
+    );
+    println!("Please copy the above text, sign it offline with 'wallet sign' cmd, and then use the signed transaction to broadcast it with 'wallet broadcast' cmd.");
+
+    Ok(())
+}
+
+async fn broadcast_signed_spends(
+    signed_tx: String,
+    client: &Client,
+    verify_store: bool,
+) -> Result<()> {
+    let (signed_spends, output_details, change_id): (
+        BTreeSet<SignedSpend>,
+        BTreeMap<UniquePubkey, (MainPubkey, DerivationIndex)>,
+        UniquePubkey,
+    ) = rmp_serde::from_slice(&hex::decode(signed_tx)?)?;
+
+    println!("The signed transaction has been successfully decoded:");
+    let mut transaction = None;
+    for (i, signed_spend) in signed_spends.iter().enumerate() {
+        println!("\nSpending input #{i}:");
+        println!("\tKey: {}", signed_spend.unique_pubkey().to_hex());
+        println!("\tAmount: {}", signed_spend.token());
+        let linked_tx = signed_spend.spent_tx();
+        if let Some(ref tx) = transaction {
+            if tx != &linked_tx {
+                bail!("Transaction seems corrupted, not all Spends (inputs) refer to the same transaction");
+            }
+        } else {
+            transaction = Some(linked_tx);
+        }
+
+        if let Err(err) = signed_spend.verify(signed_spend.spent_tx_hash()) {
+            bail!("Transaction is invalid: {err:?}");
+        }
+    }
+
+    let tx = if let Some(tx) = transaction {
+        for (i, output) in tx.outputs.iter().enumerate() {
+            println!("\nOutput #{i}:");
+            println!("\tKey: {}", output.unique_pubkey.to_hex());
+            println!("\tAmount: {}", output.amount);
+        }
+        tx
+    } else {
+        bail!("Transaction is corrupted, no transaction information found.");
+    };
+
+    use dialoguer::Confirm;
+
+    println!("\n** Please make sure the above information is correct before broadcasting it. **\n");
+    let confirmation = Confirm::new()
+        .with_prompt("Do you want to broadcast the above transaction?")
+        .interact()?;
+
+    if !confirmation {
+        println!("Transaction was not broadcasted.");
```

Comment:
then shall something to be printed to allow other manual further steps to be carried out ?
or if not `broadcasted`, then the created `tx` is no use ?

Reply:
i'll add some suggestion

Reply:
> this is when the user decided to not broadcast the tx

yeah, I know.
but in that case, the user doesn't get anything from this function.
then I don't get why the user choose to do so (the function just carried out signing but the output is for no use).

Reply:
> the user doesn't get anything from this function.
> then I don't get why the user choose to do so

the user decided to cancel the operation, that's all, it's the user's decision and here is nothing to do.